src/core_editor/line_buffer.rs

```diff
@@ -578,4 +619,87 @@ mod test {
 
         assert_eq!(4, line_buffer.word_count());
     }
+
+    #[rstest]
+    #[case("This is a test", (0, 13), "This is a tets", (0, 14))]
+    #[case("This is a test", (0, 14), "This is a test", (0, 14))]
+    // FIXME: swap_graphemes does not work with first index
+    // #[case("This is a test", (0, 0), "hTis is a test", (0, 0))]
```

Comment:
- This function is a no-op on the first location?
- Which direction should this work in? (left/right or have different methods for each)?
- How should the cursor behave on this one?

Reply:
the best way is to check against emacs and/or a shell with emacs bindings